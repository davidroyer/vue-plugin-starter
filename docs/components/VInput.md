# v-input 

The custom HTML `<input>` component. 

- **example** - <div> <v-input id="inputExample" label="My Input Example"></v-input></div> 

## props 

- `type` ***String*** (*optional*) `default: 'text'` 

- `v-model` ***String|Number*** (*optional*) 

   Use this directive to create two-way data bindings with the component. It automatically picks the correct way to update the element based on the input type. 

- `label` ***String*** (*optional*) `default: ''` 

   Provide a label 

## computed properties 

- `listeners` 

   Responsible for emitting all events from an inner element to the parent. The input method ensures that v-model works with the parent. 

   **dependencies:** `$listeners`, `$emit` 


## methods 

- `setFocus()` 

   Sets focus on input. Can be called programatically by setting an a ref and then calling this method 

