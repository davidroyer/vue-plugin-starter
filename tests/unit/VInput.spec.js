import { mount } from "@vue/test-utils";
import VInput from "@/components/VInput";

describe("VInput.vue", () => {
  it("renders component", () => {
    const wrapper = mount(VInput);
    expect(wrapper.element).toMatchSnapshot();
  });

  it("renders props.value when passed", () => {
    const wrapper = mount(VInput, {
      propsData: { value: "Initial input value" }
    });
    console.log(wrapper.vm);
    expect(wrapper.vm.value).toBe("Initial input value");
    // const inputEl = wrapper.find('.v-input')
    // expect(wrapper.text()).toEqual(value);
  });
});
