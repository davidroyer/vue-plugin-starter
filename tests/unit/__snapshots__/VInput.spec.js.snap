// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VInput.vue renders component 1`] = `
<div
  class="v-input-wrapper"
>
  <label />
   
  <input
    class="v-input"
    type="text"
  />
</div>
`;
